ASM/c/triforce.c

```diff
@@ -7,34 +7,130 @@ static uint32_t render_triforce_flag = 0;
 #define TRIFORCE_FRAMES_VISIBLE 100 // 20 Frames seems to be about 1 second
 #define TRIFORCE_FRAMES_FADE_AWAY 80
 #define TRIFORCE_FRAMES_FADE_INTO 5
+uint8_t minimap_triforce_state = 0;
 
 void set_triforce_render() {
     render_triforce_flag = 1;
     frames = frames > TRIFORCE_FRAMES_FADE_INTO ? TRIFORCE_FRAMES_FADE_INTO : frames;
 }
 
+void handle_lbutton_and_minimap_state()
+{
+    switch (minimap_triforce_state)
+    {
+        case MINIMAP_ON_SCREEN:
+            R_MINIMAP_DISABLED = render_triforce_flag == 1;
+            break;
+        case TRIFORCE_OR_SKULL_ON_SCREEN:
+            R_MINIMAP_DISABLED = 1;
+            break;
+        case NONE_ON_SCREEN:
+            R_MINIMAP_DISABLED = 1;
+            break;
+        default:
+            break;
+    }
+
+    if (z64_game.common.input[0].pad_pressed.l) {
+        minimap_triforce_state++;
+        if (minimap_triforce_state > NONE_ON_SCREEN) {
+            minimap_triforce_state = MINIMAP_ON_SCREEN;
+        }
+        switch (minimap_triforce_state)
```

Comment:
Why are there two switches on `minimap_triforce_state`. Why not do everything in one? Particularly why not set `minimap_triforce_state` in the cases themselves so you don't need to do a check beforehand due to blind increment? If anything it feels like the increment and check should go after a single switch block.

Reply:
Hmm probably because i wrote first the "press L" case, then noticed the minimap state returned to its default "on screen" value every time you load a new scene or reload, and added the first switch to keep the state persistent between areas. But you're right, this could be done in a single switch by moving checking the L presses directly in the first switch. I'll do the modification !